src/services/SessionRequestProcessor.ts

```diff
@@ -0,0 +1,213 @@
+import { APIGatewayProxyEvent } from "aws-lambda";
+import { Metrics, MetricUnits } from "@aws-lambda-powertools/metrics";
+import { Logger } from "@aws-lambda-powertools/logger";
+import { BavService } from "./BavService";
+import { HttpCodesEnum } from "../models/enums/HttpCodesEnum";
+import { MessageCodes } from "../models/enums/MessageCodes";
+import { ISessionItem } from "../models/ISessionItem";
+import { JwtPayload, Jwt } from "../models/IVeriCredential";
+import { EnvironmentVariables } from "../utils/Constants";
+import { absoluteTimeNow } from "../utils/DateTimeUtils";
+import { createDynamoDbClient } from "../utils/DynamoDBFactory";
+import { checkEnvironmentVariable } from "../utils/EnvironmentVariables";
+import { KmsJwtAdapter } from "../utils/KmsJwtAdapter";
+import { Response, UnauthorizedResponse, SECURITY_HEADERS } from "../utils/Response";
+import { buildCoreEventFields } from "../utils/TxmaEvent";
+import { isJwtValid, isPersonNameValid } from "../utils/Validations";
+
+interface ClientConfig {
+	jwksEndpoint: string;
+	clientId: string;
+	redirectUri: string;
+}
+
+export class SessionRequestProcessor {
+  private static instance: SessionRequestProcessor;
+
+  private readonly logger: Logger;
+
+	private readonly clientConfig: string;
+
+	private readonly authSessionTtlInSecs: string;
+
+	private readonly issuer: string;
+
+	private readonly txmaQueueUrl: string;
+
+	private readonly personIdentityTableName: string;
+
+  private readonly metrics: Metrics;
+
+  private readonly BavService: BavService;
+
+  private readonly kmsDecryptor: KmsJwtAdapter;
+
+  constructor(logger: Logger, metrics: Metrics) {
+  	this.logger = logger;
+  	this.metrics = metrics;
+  	logger.debug("metrics is  " + JSON.stringify(this.metrics));
+  	this.metrics.addMetric("Called", MetricUnits.Count, 1);
+
+  	const sessionTableName: string = checkEnvironmentVariable(EnvironmentVariables.SESSION_TABLE, this.logger);
```

Comment:
I thought we were gonna do this in the handler and not the processor?

Reply:
Hmm I thought it was going to be the processor because that's where they're needed - in _theory_ our handlers could trigger multiple processors different, right? Or if that were the case would we use a step function instead?